examples/pennylane_run_doubly_stochastic.py

```diff
@@ -0,0 +1,312 @@
+expectation values.
+
+Putting these two results together, `Sweke et al. (2019) <https://arxiv.org/abs/1910.01155>`__
+show that samples of the expectation value fed into the parameter-shift rule provide
+unbiased estimators of the quantum gradient---resulting in a form of stochastic gradient descent.
+Moreover, they show that convergence of the stochastic gradient descent is guaranteed
+to be on par with standard gradient descent, even in the case where the number
+of shots is 1!
+
+.. note::
+
+    It is worth noting that the smaller the number of shots used, the larger the
+    variance in the estimated expectation value. As a result, it may take
+    more optimization steps for convergence than using a larger number of shots,
+    or an exact value.
+
+    At the same time, a reduced number of shots may significantly reduce the
+    wall time of each optimization step, leading to a reduction in the overall
+    optimization time.
+
+Let's consider a simple example in PennyLane, comparing analytic gradient
+descent (with exact expectation values) to stochastic gradient descent
+using a finite number of shots.
+
+"""
+
+##############################################################################
+# A single-shot stochastic gradient descent
+# -----------------------------------------
+#
+# Consider the Hamiltonian
+#
+# .. math::
+#
+#     H = \begin{bmatrix}
+#           8 & 4 & 0 & -6\\
+#           4 & 0 & 4 & 0\\
+#           0 & 4 & 8 & 0\\
+#           -6 & 0 & 0 & 0
+#         \end{bmatrix}.
+#
+# We can solve for the ground state energy using
+# the variational quantum eigensolver (VQE) algorithm.
+#
+# Let's use the ``default.qubit`` simulator for both the analytic gradient,
+# as well as the estimated gradient using number of shots :math:`N\in\{1, 100\}`.
+
+import pennylane as qml
+from pennylane import numpy as np
+
+np.random.seed(2)
+
+from pennylane import expval
+from pennylane.init import strong_ent_layers_uniform
+from pennylane.templates.layers import StronglyEntanglingLayers
+
+num_layers = 2
+num_wires = 2
+eta = 0.01
+steps = 200
+
+dev_GD = qml.device("default.qubit", wires=num_wires, analytic=True)
+dev_SGD = qml.device("default.qubit", wires=num_wires, analytic=False)
+
+##############################################################################
+# We can use ``qml.Hermitian`` to directly specify that we want to measure
+# the expectation value of the matrix :math:`H`:
+
+H = np.array([[8, 4, 0, -6], [4, 0, 4, 0], [0, 4, 8, 0], [-6, 0, 0, 0]])
+
+
+def circuit(params):
+    StronglyEntanglingLayers(*params, wires=[0, 1])
+    return expval(qml.Hermitian(H, wires=[0, 1]))
+
+
+##############################################################################
+# Now, we create three QNodes, each corresponding to a device above,
+# and optimize them using gradient descent via the parameter-shift rule.
+
+qnode_GD = qml.QNode(circuit, dev_GD)
+qnode_SGD = qml.QNode(circuit, dev_SGD)

```

Comment:
Maybe add a `qnode_eval` here which is equal to `qnode_GD` to evaluate the cost function separately. I would keep the naming `qnode_GD` and `qnode_SGD` but I also would like to have an analytic qnode for evaluation.